llvm/include/llvm/IR/CmpPredicate.h

```diff
@@ -41,6 +41,10 @@ class CmpPredicate {
   /// Query samesign information, for optimizations.
   bool hasSameSign() const { return HasSameSign; }
 
+  /// Drops samesign information. This is used when the samesign information
+  /// should be dropped explicitly.
+  CmpInst::Predicate withoutSameSign() const { return Pred; }
```

Comment:
Possibly dropSameSign is a better name?

Reply:
Not strong opinion, but the reason I suggested withoutSameSign is that it makes it clear that this is not an in-place operation (what dropSameSign normally implies).

Reply:
I don't think this is a problem in practice, as we already have things like SmallVector::drop_front().